### PR TITLE
Add support for ubuntu focal fossa

### PR DESCRIPTION
**Pull request by Happy2C0de**
This pull requests adds support for Ubuntu Focal Fossa. 

The images were built and tested.

The GPU support is not yet complete because CUDA hasn't yet released 20.04 packages. Nevertheless, 18.04 packages can be installed. Test is missing but we should update the file as soon as possible to 20.04 CUDA. 
Update includes CUDA 10.2 (instead of 10.1) and nvidia-driver-440 (instead of nvidia-driver-418).

### CODE REVIEW

**Comment by valerytschopp** [comment]
The files:
 * `env_source/ubuntufocalfossa_SOURCE`
 * `env_source/ubuntufocalfossagpu_SOURCE` 

should be named without the `fossa` part:

 * `env_source/ubuntufocal_SOURCE`
 * `env_source/ubuntufocalgpu_SOURCE`

to be streamlined with the other ubuntu's SOURCE files...

File: diskimage_builder/elements/ubuntugpu/install.d/51-gpu [hunk 1]
```diff
@@ -2,8 +2,8 @@
 
 # Installation of general packages...
 apt-get -y install linux-headers-generic
-apt-get -y install python-pip git software-properties-common python-virtualenv
-python -m pip install --upgrade pip
+apt-get -y install python3-pip git software-properties-common python3-virtualenv
+python3 -m pip install --upgrade pip
```
Comment (valerytschopp) [inline diskimage_builder/elements/ubuntugpu/install.d/51-gpu:6]: `pip3 install --upgrade pip` ?

File: diskimage_builder/elements/ubuntugpu/install.d/51-gpu [hunk 2]
```diff
@@ -73,4 +85,4 @@ libnvinfer-dev=5.1.5-1+cuda10.0
 
 # Install tensorflow-gpu 
 # --> There is a need to convert v1.X code to v2.X. See: https://www.tensorflow.org/guide/migrate
-pip install tensorflow-gpu
+pip install --upgrade tensorflow
```
Comment (valerytschopp) [inline diskimage_builder/elements/ubuntugpu/install.d/51-gpu:88]: You've installed `python3-pip`, not sure `pip` (python3) is available?!?

`pip3 install --upgrade tensorflow`

**Review by valerytschopp — CHANGES_REQUESTED** [review]

File: diskimage_builder/elements/ubuntugpu/install.d/51-gpu (hunk 1 above)
Comment (Happy2C0de) [reply diskimage_builder/elements/ubuntugpu/install.d/51-gpu:6]: Pip3 doesn't exist. 
I checked it
`which pip` references /usr/bin/pip which is a wrapper for a call of the pip module using `#!/usr/bin/python3`
which python3 references to `#!/usr/bin/python3`

Comment (Happy2C0de) [reply diskimage_builder/elements/ubuntugpu/install.d/51-gpu:6]: I'll check the installation again.

File: env_source/ubuntufocalfossa_SOURCE [hunk 3]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
```
Comment (valerytschopp) [inline env_source/ubuntufocalfossa_SOURCE]: filename: `env_source/ubuntufocal_SOURCE`

File: env_source/ubuntufocalfossagpu_SOURCE [hunk 4]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
```
Comment (valerytschopp) [inline env_source/ubuntufocalfossagpu_SOURCE]: filename: `env_source/ubuntufocalgpu_SOURCE`

**Review by valerytschopp — CHANGES_REQUESTED** [review]
To not forget the filename without fossa...

File: diskimage_builder/elements/ubuntugpu/install.d/51-gpu (hunk 1 above)
Comment (valerytschopp) [reply diskimage_builder/elements/ubuntugpu/install.d/51-gpu:6]: On Bionic, I have this:
```
root@vt-bionic:/home/ubuntu# pip --version
pip 9.0.1 from /usr/lib/python2.7/dist-packages (python 2.7)
root@vt-bionic:/home/ubuntu# pip3 --version
pip 9.0.1 from /usr/lib/python3/dist-packages (python 3.6)
```

File: env_source/ubuntufocalfossagpu_SOURCE (hunk 4 above)
Comment (Happy2C0de) [reply env_source/ubuntufocalfossagpu_SOURCE]: Yes, this is already in the pipeline and locally changed. This needs also an adjustment on ansible repo.

I just try to run the gpu file in the ubuntu focal (basic) and check about the pip/python issue

File: diskimage_builder/elements/ubuntugpu/install.d/51-gpu (hunk 1 above)
Comment (Happy2C0de) [reply diskimage_builder/elements/ubuntugpu/install.d/51-gpu:6]: On focal, I have the selection:
```
root@focal:/home/ubuntu# python
python2    python2.7  python3    python3.8
```
So there is no /usr/bin/python anymore (which was mapped to 2.7 as far as I know)

Comment (Happy2C0de) [reply diskimage_builder/elements/ubuntugpu/install.d/51-gpu:6]: Installing `apt-get -y install python3-pip`
leads to only having a pip3 (no pip)

Then calling:
```
root@focal:/home/ubuntu# python3 -m pip install --upgrade pip
Collecting pip
  Downloading pip-20.1-py2.py3-none-any.whl (1.5 MB)
...
Installing collected packages: pip
  Attempting uninstall: pip
    Found existing installation: pip 20.0.2
    Not uninstalling pip at /usr/lib/python3/dist-packages, outside environment /usr
    Can't uninstall 'pip'. No files were found to uninstall.
Successfully installed pip-20.1
```

Comment (Happy2C0de) [reply diskimage_builder/elements/ubuntugpu/install.d/51-gpu:6]: The pip leads to:
```
root@focal:/home/ubuntu# which pip
/usr/local/bin/pip
root@focal:/home/ubuntu# cat /usr/local/bin/pip
#!/usr/bin/python3
# -*- coding: utf-8 -*-
import re
import sys
from pip._internal.cli.main import main
if __name__ == '__main__':
    sys.argv[0] = re.sub(r'(-script\.pyw|\.exe)?$', '', sys.argv[0])
    sys.exit(main())
```

Comment (valerytschopp) [reply diskimage_builder/elements/ubuntugpu/install.d/51-gpu:6]: Ok should work then

**Review by valerytschopp — APPROVED** [review]

**Pull request merged**